### ISSUE TITLE
test_pop3 flaky with randomized hashes

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5751]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)| @pitrou reported|
|-|-|
|Trac ID|trac#5751|
|Type|defect|
|Created|2012-07-08 22:59:59Z|

The following tests depend on hash ordering and fails sporadically with ``python -R``:

```
[FAIL]
Traceback (most recent call last):
  File "/home/antoine/twisted/twisted/mail/test/test_pop3.py", line 386, in _cbTestAuthListing
    ["AUTH1", "SECONDAUTH", "AUTHLAST", "."])
  File "/home/antoine/twisted/twisted/trial/unittest.py", line 270, in assertEqual
    % (msg, pformat(first), pformat(second)))
twisted.trial.unittest.FailTest: not equal:
a = ['AUTHLAST', 'SECONDAUTH', 'AUTH1', '.']
b = ['AUTH1', 'SECONDAUTH', 'AUTHLAST', '.']


twisted.mail.test.test_pop3.AnotherPOP3TestCase.testAuthListing
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/home/antoine/twisted/twisted/mail/test/test_pop3.py", line 491, in testSASL
    self.caps, self.pcaps, self.lpcaps
  File "/home/antoine/twisted/twisted/mail/test/test_pop3.py", line 467, in contained
    self.assertIn(s, c)
twisted.trial.unittest.FailTest: 'SASL SCHEME_1 SCHEME_2' not in ['TOP', 'USER', 'UIDL', 'PIPELINE', 'CELERITY', 'AUSPEX', 'POTENCE', 'IMPLEMENTATION Test Implementation String', 'EXPIRE 60 USER', 'LOGIN-DELAY 120 USER', 'SASL SCHEME_2 SCHEME_1']
```

Attachments:

* [randomhash_test_pop3.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/4b5/4b5d41fedc47db30ea1c99ca74c976925e38c3a2/ebccf3dbbf09360efbb8d017081423b66c47c0ca.patch) (1210 bytes) - added by antoine on 2012-07-08 23:00:19Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5751 5751
type__defect defect
reporter__antoine antoine
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1341788399000000 1341788399000000
changetime__1341829574000000 1341829574000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)<a name="note_1"></a>|@pitrou commented|
|-|-|

Patch attached. I couldn't import t.p.util as "util" since it would shadow t.trial.util, so I imported OrderedDict directly instead.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @exarkun|
|-|-|

Thanks.  Looks good.  I'll apply.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [34712]) Change the POP3 server tests to be independent of dict item ordering when verifying SASL capabilities.

Author: antoine
Reviewer: exarkun
Fixes: [#5751](https://github.com/twisted/twisted/issues/5751)
